### ISSUE TITLE
Developer devID return empty array

### ISSUE THREAD

**Issue by xenio** [facundoolano/google-play-scraper#319]
I am getting an empty array on developer: devID

gplay.developer({devId: "Google LLC"}).then(console.log);
return empty array []

debugging the app I get the correct url but the array is empty:

```bash
DEBUG=* node app.js
google-play-scraper Making request: {"url":"https://play.google.com/store/apps/developer?id=Google%20LLC&hl=en&gl=us"} +0ms
google-play-scraper Request finished +1s
[]
```

I am trying with 6.2.3 till 6.2.6 but with no luck
Any Idea how to fix it?

**Comment by orenbaranes** [comment]
@xenio any chance you were able to fix it? this is the only thing that does't work for me.
there was an issue with the devId encoding but that i've fixed locally.
Thanks

**Comment by xenio** [comment]
I didn't find any fix :( at the moment I am using a very old backup of the node_modules folder with old dependencies that is working.
How did you fixed the encoding locally?

**Comment by orenbaranes** [comment]
So it appears that the issue was not with the encoding but with the old url
being used (two words of developer id were not working for me)
I'm now able to fetch the first 20 apps for a developer page but struggle
to go beyond these 20. i'm new to node js so working slowly :-)

**Comment by xenio** [comment]
@orenbaranes My developer ID is with two words, and the array returned is always empty, could you please post how do you fixed?

**Comment by orenbaranes** [comment]
please see bellow:
i've used the search page as suggested and modified the URL and the
parameters a bit.

'use strict';

const request = require('./utils/request');
const scriptData = require('./utils/scriptData');
const appList = require('./utils/appList');
const R = require('ramda');

const body =
'f.req=%5B%5B%5B%22qnKhOb%22%2C%22%5B%5Bnull%2C%5B%5B10%2C%5B10%2C50%5D%5D%2Ctrue%2Cnull%2C%5B96%2C27%2C4%2C8%2C57%2C30%2C110%2C79%2C11%2C16%2C49%2C1%2C3%2C9%2C12%2C104%2C55%2C56%2C51%2C10%2C34%2C77%5D%5D%2Cnull%2C%5C%22%token%%5C%22%5D%5D%22%2Cnull%2C%22generic%22%5D%5D%5D'
;

/*
* Extract navigation tokens for next pages, parse results and call
* `checkFinished` to repeat the process with next page if necessary.
*/
function processAndRecur (html, opts, savedApps, mappings) {
if (R.is(String, html)) {
html = scriptData.parse(html);
}

const apps = appList.extract(mappings.apps, html);
const token = R.path(mappings.token, html);

return checkFinished(opts, [...savedApps, ...apps], token);
}

/*
* If already have requested results or there are no more pages, return
current
* app list, otherwise request the ajax endpoint of the next page and process
* the results.
*/
function checkFinished (opts, savedApps, nextToken) {
if (savedApps.length >= opts.num || !nextToken) {
return savedApps.slice(0, opts.num);
}

const requestOptions = Object.assign({
url: '
https://play.google.com/_/PlayStoreUi/data/batchexecute?rpcids=qnKhOb&bl=boq_playuiserver_20190424.04_p0&hl=en&gl=us&authuser=0&soc-app=121&soc-platform=1&soc-device=1
',
method: 'POST',
body: body.replace('%token%', nextToken),
followAllRedirects: true,
headers: {
'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
}
}, opts.requestOptions);

return request(requestOptions, opts.throttle)
.then((html) => {
const input = JSON.parse(html.substring(5));
const data = JSON.parse(input[0][2]);
return processAndRecur(data, opts, savedApps, REQUEST_MAPPINGS);
});
}

/*
* Make the first search request as in the browser and call `checkfinished`
to
* process the next pages.
*/
function initialRequest (opts) {
// sometimes the first result page is a cluster of subsections,
// need to skip to the full results page
function skipClusterPage (html) {
const match = html.match(/href="\/store\/apps\/collection\/
search_collection_more_results_cluster?(.*?)"/);
if (match) {
const innerUrl = 'https://play.google.com/' + match[0].split(/"/)[1];
return request(Object.assign({
url: innerUrl
}, opts.requestOptions), opts.throttle);
}
return html;
}

//const url = `
https://play.google.com/store/search?c=apps&q=${opts.term}&hl=${opts.lang}&gl=${opts.country}&price=${opts.price}`
;
const url = `https://play.google.com/store/apps/developer?id=${opts.devId}
&hl=${opts.lang}&gl=${opts.country}`;
return request(Object.assign({url}, opts.requestOptions), opts.throttle)
.then(skipClusterPage)
.then((html) => processAndRecur(html, opts, [], INITIAL_MAPPINGS));
}

const INITIAL_MAPPINGS = {
apps: ['ds:3', 0, 1, 0, 0, 0],
token: ['ds:3', 0, 1, 0, 0, 7, 1]
};

const REQUEST_MAPPINGS = {
apps: [0, 0, 0],
token: [0, 0, 7, 1]
};

function getPriceGoogleValue (value) {
switch (value.toLowerCase()) {
case 'free':
return 1;
case 'paid':
return 2;
case 'all':
default:
return 0;
}
}

function developer (getParseList, opts) {
return new Promise(function (resolve, reject) {
if (!opts || !opts.devId) {
throw Error('DevId is missing');
}

if (opts.num && opts.num > 250) {
throw Error("The number of results can't exceed 250");
}

opts = {
devId: encodeURIComponent(opts.devId),
lang: opts.lang || 'en',
country: opts.country || 'us',
num: opts.num || 20,
fullDetail: opts.fullDetail,
price: opts.price ? getPriceGoogleValue(opts.price) : 0,
throttle: opts.throttle,
cache: opts.cache,
getParseList,
requestOptions: opts.requestOptions
};

initialRequest(opts)
.then(resolve)
.catch(reject);
});
}

module.exports = developer;

**Comment by xenio** [comment]
Thanks @orenbaranes your temporary fix is working (only 20 results).

**Comment by facundoolano** [comment]
fixed by #344
